docs/courses/README.md

```diff
@@ -1,28 +1,104 @@
+
+When you enter a course page, for example the Spanish course [here](https://librelingo.app/course/spanish-from-english/), you'll see headings like Basics, Introduction, Activities. These are modules. And under each module heading there are a list of skills under it. Modules are merely there to serve the purpose of organizing skills and basically nothing else. Think of them as folders for your skills.
+
+"Skills" are the actual lessons focused on a particular topic. Each skill is like a set of questions that follow a particular theme. For example, "Animal" skill has questions relating to animals, "Food" skill has questions relating to foods. Each skill is designed to test the skill on a certain topic.
+
+Inside a skill YAML file there are some keys that you might want to learn of, which is exaplained in detail [here](course.md).
+
+<a id="things-new-contributors"></a>
+### Things you can do as a new contributor
+
+If you are new in contributing to course data, these are the things you can do to get a proper understanding of the courses and have a good time with the project:
+
+- Go to [LibreLingo's Spanish course](https://librelingo.app/course/spanish-from-english) and try out some of the skills. Just get an idea on how the questions are structured and the type of answers made available.
+- Clone this repository or [download it](https://github.com/kantord/LibreLingo/archive/main.zip) and study the `courses` directory and how everything's laid out. Be sure to have a good text editor, such as [Notepad++](https://notepad-plus-plus.org/), [Geany](https://geany.org/) in hand to view or edit files.
+- Try to learn how data is organized within .yaml files and directories. [Course](course.md), [Module](module.md) and [Skill](skill.md) documentation may help you on this.
+- Try to learn how YAML works. (see [here](https://en.wikipedia.org/wiki/YAML#Syntax) or [here](https://blog.stackpath.com/yaml/) or [here](https://www.cloudbees.com/blog/yaml-tutorial-everything-you-need-get-started/))
+- Determine some information about your language.
+  - Your language name in English. e.g. French, Ancient Greek
+  - Language slug (lowercase and no spaces). e.g. french, ancient-greek
+  - IETF BCP 47 tag name for your language, which you can get from [here](http://www.iana.org/assignments/language-subtag-registry). Just look for the `Subtag` for your language. e.g. fr, grc
+- If you have questions, feel free to ask on [Discussions](https://github.com/kantord/LibreLingo/discussions).
+
+
+<a id="courses-open"></a>
 ## Courses currently open for contributions
 
+<a id="courses-from-english"></a>
```

Comment:
I think these should probably not be necessary, right? I mean the `<a />` tags. I think most markdown processors add those automatically

Reply:
Yes, but the problem is that they're automatic and long.
- If we ever need to edit a title, it will change the existing anchors which would break existing links already shared. Editing all those occurrences would be a chore and sometimes old messages cannot be edited.
- It gives us control how we want our anchors to be, which word to include. For example, "Courses currently open for contributions" would give us "#courses-currently-open-for-contributions" which is long and unnecessary. Instead we can have "#courses-open" which is friendlier and short.

I saw it first on [this manual](https://github.com/void-linux/void-packages/blob/master/Manual.md) and I found that this is useful in reference documentations. One thing however is that when sharing, the links under "Table of contents" should be shared, not the ones present on the shackle icon. Otherwise it would defeat the purpose.

Reply:
ok, that makes sense I think! 

I don't think we can fully avoid breaking changes, but I can see that it makes sense to at least try